torch/ao/quantization/experimental/observer.py

```diff
@@ -4,44 +4,135 @@
 """
 
 import torch
+import itertools
 from torch.ao.quantization.observer import ObserverBase
 from typing import Tuple
 
 class NonUniformQuantizationObserverBase(ObserverBase):
-    quant_min = None
-    quant_max = None
+    min_val: torch.tensor([])
+    max_val: torch.tensor([])
+    level_indices: torch.tensor([])
+    b: int
+    k: int
+    n: int
+    alpha: float
+    gamma: float
 
     def __init__(
         self,
-        min_val: torch.Tensor,
-        max_val: torch.Tensor,
-        level_indices: torch.Tensor,
-            b: int,
-            k: int) -> None:
-        super().__init__
+        min_val=None,
+        max_val=None,
+        b=0,
+        k=0,
+            dtype=torch.quint8) -> None:
+        super().__init__(dtype)
+        self.min_val = min_val
+        self.max_val = max_val
+        self.level_indices = torch.tensor([])
+        self.b = b
+        self.k = k
+        self.n = 0
```

Comment:
generally a bit safer to initialize to None rather than a valid number if you want a placeholder value, both for k and n